numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -730,6 +732,24 @@ VOID_getitem(void *input, void *vap)
         return (PyObject *)ret;
     }
 
+    if (!future_warning_emitted){
```

Comment:
Hmm, Is this restriction needed? For that matter, is it wanted? What if someone wants to find all the places that need fixing?